### ISSUE TITLE
Create a rule to forbid subclassing a parent class

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#9731]
## Expected Behavior of the rule
If set for `androidx.lifecycle.ViewModel` then below code will be reported by the detekt

```kt
class CustomVm(
) : androidx.lifecycle.ViewModel() {
```

## Context
This can be useful, when a project defines something custom which is different then what is recommended by the community
